src/mmtk.rs

```diff
@@ -597,3 +597,31 @@ impl<VM: VMBinding> MMTK<VM> {
             .initialize_object_metadata(object, false)
     }
 }
+
+/// A non-mangled function to print object information for debugging purposes. This function can be directly
+/// called from a debugger.
+#[no_mangle]
+pub fn mmtk_debug_print_object_info(object: crate::util::ObjectReference) {
+    // If the address is unmapped, we cannot access its metadata. Just quit.
+    assert!(
+        object.to_raw_address().is_mapped(),
+        "{} is not mapped in MMTk",
+        object
+    );
```

Comment:
This is an assertion.  If this fails, it will panic.  It is right for normal execution, but this function is for calling in GDB.  For example,

```gdb
(gdb) p (int)abort()

Program received signal SIGABRT, Aborted.
0x00007ffff7e0174c in ?? () from /usr/lib/libc.so.6
The program being debugged was signaled while in a function called from GDB.
GDB remains in the frame where the signal was received.
To change this behavior use "set unwind-on-signal on".
Evaluation of the expression containing the function
(abort) will be abandoned.
When the function is done executing, GDB will silently stop.
(gdb) 
```

It is better to just use `println!` to print an error message, and return if it is impossible to proceed.

Reply:
Done.